cluster-assign-roles.sh

```diff
@@ -232,7 +238,13 @@ function kafka_install {
     done
 
     printf "Installing kafka zookeeper heads...\n"
-    install_machines $(printf ${hosts// /\\n} | grep -i "BCPC-Kafka-Head-Zookeeper" | sort)
+    for m in $(printf ${hosts// /\\n} | grep -i "BCPC-Kafka-Head-Zookeeper" | sort); do
+      [[ "$m" =~ $REGEX ]]
+      # authenticate the node one by one
+      vaults=$(sudo ./find_resources.rb ${BASH_REMATCH[3]} | tail -1)
+      sudo ./node_auth.rb $vaults ${BASH_REMATCH[3]}
```

Comment:
Since we twice use `${BASH_REMATCH[3]}` could you set this to a descriptive variable? (E.g. nodename?)